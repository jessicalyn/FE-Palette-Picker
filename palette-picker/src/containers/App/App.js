import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchProjects } from '../../thunks/fetchProjects';

export class App extends Component {

  componentDidMount() {
    const url = `https://palette-picker-mfjk.herokuapp.com/api/v1/projects`
    this.props.fetchProjects(url)
  }

  render() {

    return (
      <div className="App">
        <header className="App-header">
          <p>Here we are</p>
        </header>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  error: state.error,
  projects: state.projects
})

export const mapDispatchToProps = (dispatch) => ({
  fetchProjects: (url) => dispatch(fetchProjects(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(App)
