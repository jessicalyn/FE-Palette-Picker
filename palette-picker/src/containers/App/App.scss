.App {
  height: 800px;
  width: 500px;
}

.colors-wrapper {
  display: flex;
  flex-direction: column;
  height: 80%;
  width: 80%;
  background-color: grey;
  border: 2px solid lime;
}

.palette-color {
  height: 15%;
  width: 80%;
  margin: auto;
  background-color: white;
}