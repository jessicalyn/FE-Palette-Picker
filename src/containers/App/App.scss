.App {
  height: 100%;
  width: 100%;
  display: flex;
  // border: 2px solid orange;
}

.header-project-wrapper {
  // border: 2px solid blue;
  flex: 1;
}

.app-header {
  // height: 30%;
  // border: 2px solid black;
}

h1 {
  margin: 30px;
  font-size: 50px;
  font-family: $font-title;
}

.projects-wrapper {
  // background-color: purple;
  border: 2px solid green;
}

.palette-wrapper {
  flex: 1;
  display: flex;
  flex-direction: column;
  width: 100%;
  border: 2px solid hotpink;
  // background-color: yellow
}

input {
  border-radius: 5px;
  border: 1px solid black;
  padding: 5px;
  margin: 10px;
  height: 25px;
  width: 200px
}

button {
  border: none;
  text-decoration: none;
  background: transparent;
  font-size: 18px;
  font-family: $font-main;
}