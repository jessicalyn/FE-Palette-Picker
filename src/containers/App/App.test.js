import { App, mapStateToProps, mapDispatchToProps } from './App';
import React from 'react';
import { shallow } from 'enzyme';

jest.mock('../../thunks/fetchProjects')
jest.mock('../../thunks/fetchPalettes')

describe('App Container', () => {
  describe('App', () => {
    let wrapper
    const mockError = "Something went wrong"
    const mockProjects = [
      { project_id: 1,
        project_name: "Travel App",
        created_at: "2019-05-10T17:45:09.612Z",
        updated_at: "2019-05-10T17:45:09.612Z"
      },
      { project_id: 2,
        project_name: "Fitness App",
        created_at: "2019-05-10T17:45:09.634Z",
        updated_at: "2019-05-10T17:45:09.634Z"
      }
    ]
    const mockCurrentColors = [
      {"color": "#f82e41", "locked": true}, 
      {"color": "#2261cd", "locked": true}, 
      {"color": "#271de3", "locked": true}, 
      {"color": "#1563ec", "locked": true}, 
      {"color": "#9e0246", "locked": true}
    ]
    let mockFetchProjects
    let mockStoreColors
    let mockFetchPalettes
    
    beforeEach(() => {
      mockFetchProjects = jest.fn().mockImplementation(() => Promise.resolve({mockProjects}))
      mockStoreColors = jest.fn().mockImplementation(() => Promise.resolve({mockCurrentColors}))
      mockFetchPalettes = jest.fn()
      wrapper = shallow(
        <App 
          error={mockError}
          projects={mockProjects}
          fetchProjects={mockFetchProjects}
          currentColors={mockCurrentColors}
          storeColors={mockStoreColors}
          fetchPalettes={mockFetchPalettes}
        />
      )
    })

    it('should match the snapshot with all data passed in', () => {
      expect(wrapper.debug()).toMatchSnapshot()
    })

    it('should invoke fetchProjects on componentDidMount', () => {
      wrapper.instance().componentDidMount()
      
      expect(mockFetchProjects).toHaveBeenCalled()

    })

    it('should invoke generateRandomColor on componentDidMount', () => {
      const spy = jest.spyOn(wrapper.instance(), 'generateRandomColor')
      
      wrapper.instance().componentDidMount()

      expect(spy).toHaveBeenCalled()
    })

    it.skip('should call props function storeColors with updatedColors array', () => {
      const mockUpdatedColors = [
        {"color": "#f82e41", "locked": true}, 
        {"color": "#2261cd", "locked": true}, 
        {"color": "#271de3", "locked": true}, 
        {"color": "#1563ec", "locked": true}, 
        {"color": "#9e0246", "locked": true}
      ]

      wrapper.instance().updateColors()

      expect(wrapper.instance().props.storeColors).toHaveBeenCalledWith(mockUpdatedColors)
    })
  });

  describe('mapStateToProps', () => {
    it('should return an object with an error string, projects array and currentColors array', () => {
      const mockState = {
        projects: [
          { project_id: 1,
            project_name: "Travel App",
            created_at: "2019-05-10T17:45:09.612Z",
            updated_at: "2019-05-10T17:45:09.612Z"
          },
          { project_id: 2,
            project_name: "Fitness App",
            created_at: "2019-05-10T17:45:09.634Z",
            updated_at: "2019-05-10T17:45:09.634Z"
          }
        ],
        error: "",
        currentColors: [
          {"color": "#f82e41", "locked": true}, 
          {"color": "#2261cd", "locked": true}
        ],
        fakeState: "Not real state to return"
      }

      const expected = {
        projects: [
          { project_id: 1,
            project_name: "Travel App",
            created_at: "2019-05-10T17:45:09.612Z",
            updated_at: "2019-05-10T17:45:09.612Z"
          },
          { project_id: 2,
            project_name: "Fitness App",
            created_at: "2019-05-10T17:45:09.634Z",
            updated_at: "2019-05-10T17:45:09.634Z"
          }
        ],
        error: "",
        currentColors: [
          {"color": "#f82e41", "locked": true}, 
          {"color": "#2261cd", "locked": true}
        ]
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  });

  describe('mapDispatchToProps', () => {
    it('should dispatch fetchProjects with url', () => {
      const mockDispatch = jest.fn()
      const fetchProjects = jest.fn()
      const actionToDispatch = fetchProjects('www.projects.com')
      const mappedProps = mapDispatchToProps(mockDispatch)
      
      mappedProps.fetchProjects('www.projects.com')

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should dispatch storeColors with an array of colors', () => {
      const mockColors = [
        {"color": "#f82e41", "locked": true}, 
        {"color": "#2261cd", "locked": true}
      ]
      const mockDispatch = jest.fn()
      const storeColors = jest.fn()
      const actionToDispatch = storeColors(mockColors)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.storeColors(mockColors)

      expect(mockDispatch).toHaveBeenCalled(actionToDispatch)
    })

    it('should dispatch fetchPalettes', () => {
      const mockDispatch = jest.fn()
      const fetchPalettes = jest.fn()
      const actionToDispatch = fetchPalettes()
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.fetchPalettes()

      expect(mockDispatch).toHaveBeenCalled(actionToDispatch)
    })

  })
});