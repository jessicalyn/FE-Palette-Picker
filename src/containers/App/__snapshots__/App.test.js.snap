// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container App should match the snapshot with all data passed in 1`] = `
"<div className=\\"App\\">
  <div className=\\"header-project-wrapper\\">
    <header className=\\"app-header\\">
      <h1>
        Palette Picker
      </h1>
    </header>
    <div className=\\"projects-wrapper\\">
      <ConnectFunction />
    </div>
  </div>
  <div className=\\"palette-wrapper\\">
    <ConnectFunction />
  </div>
</div>"
`;
